controllers/datadogagent/feature/externalmetrics/feature_test.go

```diff
@@ -85,30 +85,36 @@ func TestExternalMetricsFeature(t *testing.T) {
 		//////////////////////////
 		{
 			Name:          "v2alpha1 external metrics not enabled",
-			DDAv2:         newV2Agent(false, false, false, v2alpha1.DatadogCredentials{}),
+			DDAv2:         newV2Agent(false, true, false, false, v2alpha1.DatadogCredentials{}),
 			WantConfigure: false,
 		},
 		{
 			Name:          "v2alpha1 external metrics enabled",
-			DDAv2:         newV2Agent(true, true, false, v2alpha1.DatadogCredentials{}),
+			DDAv2:         newV2Agent(true, true, true, false, v2alpha1.DatadogCredentials{}),
 			WantConfigure: true,
 			ClusterAgent:  testDCAResources(true, false, false),
 		},
 		{
 			Name:          "v2alpha1 external metrics enabled, wpa controller enabled",
-			DDAv2:         newV2Agent(true, true, true, v2alpha1.DatadogCredentials{}),
+			DDAv2:         newV2Agent(true, true, true, true, v2alpha1.DatadogCredentials{}),
 			WantConfigure: true,
 			ClusterAgent:  testDCAResources(true, true, false),
 		},
 		{
 			Name:          "v2alpha1 external metrics enabled, ddm disabled",
-			DDAv2:         newV2Agent(true, false, false, v2alpha1.DatadogCredentials{}),
+			DDAv2:         newV2Agent(true, true, false, false, v2alpha1.DatadogCredentials{}),
 			WantConfigure: true,
 			ClusterAgent:  testDCAResources(false, false, false),
 		},
 		{
 			Name:          "v2alpha1 external metrics enabled, secrets set",
-			DDAv2:         newV2Agent(true, true, false, secretV2),
+			DDAv2:         newV2Agent(true, true, true, false, secretV2),
+			WantConfigure: true,
+			ClusterAgent:  testDCAResources(true, false, true),
```

Comment:
This and next test pass different values for `registerAPIService` but make same assertions using`testDCAResources(true, false, true)` which means `testDCAResources` is probably not covering resources what's inside `if f.registerAPIService {..`. Could you please fix that?